python/lsst/daf/butler/transfers/_context.py

```diff
@@ -155,14 +150,16 @@ def saveDatasets(self, refs: Iterable[DatasetRef], *,
                 exports = [rewrite(export) for export in exports]
             self._dataset_ids.add(ref.getCheckedId())
             assert ref.run is not None
-            datasets[ref.datasetType, ref.run].extend(exports)
+            self._datasets[ref.datasetType, ref.run].extend(exports)
         self.saveDataIds(dataIds, elements=elements)
-        for (datasetType, run), records in datasets.items():
-            self._backend.saveDatasets(datasetType, run, *records)
 
     def _finish(self) -> None:
         """Delegate to the backend to finish the export process.
 
         For use by `Butler.export` only.
         """
+        for element in self._registry.dimensions.sorted(self._records.keys()):
+            self._backend.saveDimensionData(element, *self._records[element].values())
+        for (datasetType, run), records in self._datasets.items():
```

Comment:
Could this benefit from a sort?

Reply:
I'll add one.  As in the previous case, we'll need to add sorting in the line below as well to make the order fully deterministic, but we might as well start.